src/node/test/channels.cpp

```diff
@@ -112,49 +136,139 @@ auto read_node_msgs(ringbuffer::Circuit& circuit)
   return std::make_pair(add_node_msgs, remove_node_msgs);
 }
 
+NodeOutboundMsg<MsgType> get_first(
+  ringbuffer::Circuit& circuit, NodeMsgType msg_type)
+{
+  auto outbound_msgs = read_outbound_msgs<MsgType>(circuit);
+  REQUIRE(outbound_msgs.size() == 1);
+  auto msg = outbound_msgs[0];
+  const auto* data_ = msg.payload.data();
+  auto size_ = msg.payload.size();
+  REQUIRE(msg.type == msg_type);
+  return msg;
+}
+
+void hexdump(const char* hdr, const std::vector<uint8_t>& data)
```

Comment:
FYI, there's a new `ds/hex.h` that should handle all conversions from/to hex strings.

Reply:
Thanks, found that and started using it in other places.